recovery_backend/backend/configmap.go

```diff
@@ -103,34 +107,36 @@ func (rb *RecoveryBackend) AddClientID(hostname, version string, clientID Client
 
 	logrus.Infof("Adding client '%v' in recovery backend %v (version %v)", clientID, configMapName, version)
 
-	cm, err := rb.Datastore.GetConfigMap(rb.Namespace, configMapName, false)
-	if err != nil {
-		return errors.Wrapf(err, "failed to get configmap %v", configMapName)
-	}
+	_, err := util.RetryOnConflictCause(func() (interface{}, error) {
+		cm, err := rb.Datastore.GetConfigMap(rb.Namespace, configMapName, false)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to get configmap %v", configMapName)
+		}
 
-	dataStr := cm.Data[version]
-	if dataStr == "" {
-		return errors.Wrapf(err, "failed to get data from recovery backend %v (version %v)", configMapName, version)
-	}
+		dataStr := cm.Data[version]
```

Comment:
NIT: Why not using format `dataStr, ok := cm.Data[version]` here?

Reply:
`""` is also a valid value for `dataStr, ok := cm.Data[version]`.